### DISCUSSION TITLE
不能通过命令获取debian12 的ipv6地址

### DISCUSSION THREAD

**Discussion by Jengro777** [jeessy2/ddns-go#786] [Q&A]
![image](https://github.com/jeessy2/ddns-go/assets/35259040/622ea91a-4280-4fe7-98c0-af322d6cd242)

**Comment by Jengro777** [comment]
执行命令的时候为什么会自动添加 `/bin/bash -rc` 呢？ 没有这个可以正常获取ipv6地址

**Comment by WaterLemons2k** [comment]
> ![image](https://user-images.githubusercontent.com/35259040/256154963-622ea91a-4280-4fe7-98c0-af322d6cd242.png)

看报错是找不到 `sudo`，去掉 `sudo` 试试。

**Reply by Jengro777** [reply]
不是 sudo 的问题，去掉也是一样

**Reply by WaterLemons2k** [reply]
[v5.4.0](https://github.com/jeessy2/ddns-go/releases/tag/v5.4.0) 后就用 `/bin/bash -c` 运行命令了，再试试 v5.4.0。

**Reply by Jengro777** [reply]
使用的5.5.0的版本
debian12

![image](https://github.com/jeessy2/ddns-go/assets/35259040/8b9f2ab4-9416-4e85-aaa9-5e8a930fc1c2)

![image](https://github.com/jeessy2/ddns-go/assets/35259040/27455d8a-ef27-4496-a5bc-5bb5f474ad80)

**Reply by WaterLemons2k** [reply]
请转义：

```sh
/bin/sh -c "ip -6 addr | grep inet6 | awk -F '[ \t]+|/' '\$3 == \"::1\" { next;} \$3 ~ /^fe80::/ { next;} /inet6/ {print \$3}'"
```

**Comment by Jengro777** [comment]
@WaterLemons2k 可以去掉默认的  `/bin/sh -c`  吗？让用户自行输入

**Reply by WaterLemons2k** [reply]
> 可以去掉默认的 `/bin/sh -c` 吗？让用户自行输入

默认的 `/bin/sh -c` 是在这里添加的：

https://github.com/jeessy2/ddns-go/blob/6bc09d799c8468a3d0f675a8f2459226b0afcfaf/config/config.go#L229-L236

**Reply by Jengro777** [reply]
要自己构建吗? 
算了，等后面修改了 还是使用官方的镜像吧！
自己懒得折腾

**Comment by Jengro777** [comment]
2023/08/01 18:45:27 监听 :9876 ...
2023/08/01 18:48:00 配置文件已保存在: /root/.ddns_go_config.yaml
2023/08/01 18:48:10 配置文件已保存在: /root/.ddns_go_config.yaml
2023/08/01 18:48:22 配置文件已保存在: /root/.ddns_go_config.yaml
2023/08/01 18:48:22 连接失败! <a target='blank' href='https://speed.neu6.edu.cn/getIP.php'>点击查看接口能否返回IPv6地址</a>, 参考说明:<a target='blank' href='https://github.com/jeessy2/ddns-go#使用ipv6'>点击访问</a>
2023/08/01 18:48:22 错误信息: Get "https://speed.neu6.edu.cn/getIP.php": dial tcp6 [2001:da8:9000::66]:443: connect: cannot assign requested address
2023/08/01 18:48:22 连接失败! <a target='blank' href='https://v6.ident.me'>点击查看接口能否返回IPv6地址</a>, 参考说明:<a target='blank' href='https://github.com/jeessy2/ddns-go#使用ipv6'>点击访问</a>
2023/08/01 18:48:22 错误信息: Get "https://v6.ident.me": dial tcp6 [2a01:4f8:c0c:bd0a::1]:443: connect: cannot assign requested address
2023/08/01 18:48:22 连接失败! <a target='blank' href='https://6.ipw.cn'>点击查看接口能否返回IPv6地址</a>, 参考说明:<a target='blank' href='https://github.com/jeessy2/ddns-go#使用ipv6'>点击访问</a>
2023/08/01 18:48:22 错误信息: Get "https://6.ipw.cn": dial tcp6 [2402:4e00:1013:e500:0:9671:f018:4947]:443: connect: cannot assign requested address
2023/08/01 18:48:22 未能获取IPv6地址, 将不会更新

**Reply by Jengro777** [reply]
已经获取到ipv6 地址，但什么还是提示未获取到ip地址呢

**Reply by WaterLemons2k** [reply]
短连接过多，导致timewait连接过多，消耗的本地端口过多，当新连接建立的时候服务器本身无法分配新端口就有 Cannot assign requested address 的日志

参考：https://www.cnblogs.com/thatsit/p/cannot-assign-requested-address.html
